logstash-core/src/test/java/org/logstash/ackedqueue/HeadPageTest.java

```diff
@@ -1,104 +1,98 @@
 package org.logstash.ackedqueue;
 
 import org.junit.Test;
-import org.logstash.ackedqueue.io.ByteBufferPageIO;
 import org.logstash.ackedqueue.io.PageIO;
 
 import java.io.IOException;
 
 import static org.hamcrest.CoreMatchers.equalTo;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.logstash.ackedqueue.QueueTestHelpers.singleElementCapacityForByteBufferPageIO;
 
 public class HeadPageTest {
 
     @Test
     public void newHeadPage() throws IOException {
         Settings s = TestSettings.volatileQueueSettings(100);
-        Queue q = new Queue(s);
-        PageIO pageIO = s.getPageIOFactory().build(0, 100, "dummy");
-        pageIO.create();
-        HeadPage p = new HeadPage(0, q, pageIO);
-
-        assertThat(p.getPageNum(), is(equalTo(0)));
-        assertThat(p.isFullyRead(), is(true));
-        assertThat(p.isFullyAcked(), is(false));
-        assertThat(p.hasSpace(10), is(true));
-        assertThat(p.hasSpace(100), is(false));
-
-        q.close();
+        try(Queue q = new Queue(s)) {
+            PageIO pageIO = s.getPageIOFactory().build(0, 100, "dummy");
+            pageIO.create();
+            HeadPage p = new HeadPage(0, q, pageIO);
```

Comment:
@robbavey I know it already was like this, but it seems we should also close `p` eventually right? It's a `Closeable` too as far as I can see. Maybe we can add another `try-with-resources` nested around the `assertThat` ?